tests/test_remote_configuration.py

```diff
@@ -34,3 +35,57 @@ def validator(data):
             assert language == expected_language, f"language is '{language}', I was expecting '{expected_language}'"
 
         interfaces.library.add_remote_configuration_validation(validator=validator, is_success_on_expiry=True)
+
+
+@coverage.basic
+class Test_RemoteConfigurationUpdateSequence(BaseTestCase):
+    """Remote configuration update sequence"""
+
+    current_test = -1
+    number_of_tests = 13
+
+    def test_tracer_update_sequence(self):
+        def load_expected_payload(test_no):
+            path = "???/r{}.json".format(test_no)
+            f = open(path, "r")
+            payload = json.load(f)
+            f.close()
+            return payload
+
+        def validator(data):
```

Comment:
I think I'm having trouble understanding how this all connects together with #366.

It sounds like I need to do something like:

```python
from utils import proxies

proxies.set("mock_remote_config_backend", True)
```

But then I'm not sure how I go about queuing up the responses. I see that there is a modify_response function within the proxy forwarder where the flag is checked and the response is modified if True, but I don't see a way to set this here in the test definition. https://github.com/DataDog/system-tests/blob/8988a10f4fb4db35ae12472351e98933a513b787/utils/proxy/forwarder.py#L27-L35

I also don't understand how to control the timing of all of this. As soon as the weblog application that embeds the tracer starts up, the remote configuration client within it will start making a request behind the scenes for an update every X seconds (X is based on how we configure it). It's my understanding that `validator` defined here is called for every request to the /v0.7/config endpoint that the proxy captures. I want the first 13 requests from the tracer to be met with 13 different responses in sequence. So for this style test to work I'd need the proxy preconfigured before the tracer starts its request polling.

Let me know if it'd be better to have a quick sync up. I currently have a Go program that runs the tracer that mimics this interception of requests that I've used in the interim, so I can show a concrete example of what we'd need.

Reply:
let's meet :)